projects/js-packages/publicize-components/src/components/add-connection-modal/index.tsx

```diff
@@ -20,100 +21,71 @@ const AddConnectionModal = ( {
 	currentService,
 	setCurrentService,
 }: AddConnectionModalProps ) => {
-	const supportedServices = useSupportedServices();
+	const [ keyringResult, setKeyringResult ] = useState< KeyringResult | null >( null );
 
 	const [ isSmall ] = useBreakpointMatch( 'sm' );
 
-	const onServiceSelected = useCallback(
-		service => () => {
-			setCurrentService( service );
-		},
-		[ setCurrentService ]
-	);
-
 	const onBackClicked = useCallback( () => {
 		setCurrentService( null );
 	}, [ setCurrentService ] );
 
-	const onConfirm = useCallback( ( data: unknown ) => {
-		// eslint-disable-next-line no-console
-		console.log( data );
+	const onConfirm = useCallback( ( result: KeyringResult ) => {
+		setKeyringResult( result );
 	}, [] );
 
+	const onComplete = useCallback( () => {
+		setKeyringResult( null );
+		onCloseModal();
+	}, [ onCloseModal ] );
+
+	const hasKeyringResult = Boolean( keyringResult?.ID );
+
+	// Use IIFE to avoid nested ternary and messed up minification
```

Comment:
Having two functions is not a problem as it makes the code more readable and does not affect performance because the value is always going to be the same for a given input.

Reply:
Ok, that's probably a question of style. It won't affect performance much, but it definitely does.